templates/user/apiProfileForm.tpl

```diff
@@ -23,21 +23,37 @@
 
 	{include file="controllers/notification/inPlaceNotification.tpl" notificationId="apiProfileNotification"}
 
-	{fbvFormSection list=true}
+	{* {fbvFormSection list=true}
 		{fbvElement id=apiKeyEnabled type="checkbox" label="user.apiKeyEnabled" checked=$apiKeyEnabled value=1}
 		{fbvElement id=generateApiKey type="checkbox" label="user.apiKey.generate" value=1}
-	{/fbvFormSection}
-	<p>{translate key="user.apiKey.generateWarning"}</p>
+	{/fbvFormSection} *}
 
-	{fbvFormSection}
+	{fbvFormSection title="user.apiKey"}
 		{if !$apiKey}{assign var=apiKey value="common.none"|translate}{/if}
-		{fbvElement id=apiKey type="text" label="user.apiKey" readonly="true" value=$apiKey size=$fbvStyles.size.MEDIUM}
+		{fbvElement id=apiKey type="text" readonly="true" inline=true value=$apiKey size=$fbvStyles.size.MEDIUM}
+		{if !$apiSecretMissing}
+			{fbvElement id=apiKeyAction type="hidden" readonly="true" value=$apiKeyAction}
+			<button
+				type="submit"
+				{if $apiKeyAction === \PKP\user\form\APIProfileForm::API_KEY_DELETE}
+					onClick="return confirm({translate|json_encode|escape key='user.apiKey.remove.confirmation.message'})"
```

Comment:
without the use of both `json_encode` and `escape` , the confirmation popup not working as I tested. thats why it's both are there .

Reply:
Hmm, interesting -- can you check the generated source to see what's causing the problem?

Reply:
`escape` is required to escape double quotes so that proper confirm message can me constructed through the translation . I have also noticed that in other places that is how it's used, for example in `lib/pkp/templates/admin/index.tpl`

```html
<button class="pkpButton pkpButton--isWarnable" onclick="return confirm({translate|json_encode|escape key="admin.confirmExpireSessions"})">{translate key="admin.expireSessions"}</button>
```

Reply:
Ah, brain fart -- `json_encode` for the javascript, and `escape` because we're inside an HTML attribute value. That's as it should be, sorry for the diversion!